File: leetcode_py/Rotate Image.py
class Solution:
    # @param matrix, a list of lists of integers
    # @return nothing (void), do not return anything, modify matrix in-place instead.
    def rotate(self, matrix):
        for i in range(len(matrix) / 2):
            for j in range(len(matrix[0])):
                matrix[len(matrix) - 1 - i][j], matrix[i][j] = matrix[i][j], matrix[len(matrix) - 1 - i][j]

        for i in range(len(matrix)):
            for j in range(i + 1, len(matrix[0])):
                matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]


class Solution:
    def rotate(self, matrix):
        return [list(reversed(x)) for x in zip(*matrix)]

File: leetcode_js/Rotate Image.ts
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */

let rotate :(matrix:number[][])=>any
 = function(matrix:number[][]) {
 	for(let i:number = 0; i < matrix.length/2; i++){
 		for(let j:number = 0; j < matrix[0].length; j++){
 			let op:number = matrix.length - 1 - i;
 			let tmp: number = matrix[op][j];
 			matrix[op][j] = matrix[i][j];
 			matrix[i][j] = tmp;
 		}
 	}
 	for(let i: number = 0; i < matrix.length; i++){
 		for(let j: number = i + 1; j < matrix[0].length; j++){
 			let tmp: number = matrix[i][j];
 			matrix[i][j] = matrix[j][i];
 			matrix[j][i] = tmp;
 		}
 	}
 }
    